src/reset-password/service.ts

```diff
@@ -43,24 +43,21 @@ export async function createToken(user: User): Promise<string> {
 
     await database.query(query, [
         user.id,
-        await bcrypt.hash(token, 12),
+        token,
         tokenTTL
     ]);
     return token;
 }
 
-export async function validateToken(user: User, token: string): Promise<boolean> {
-    const query = 'SELECT token FROM reset_password_token WHERE user_id = ? AND token = ? AND expires_at > UNIX_TIMESTAMP()';
-    const result = await database.query(query, [user.id]);
+export async function validateToken(token: string): Promise<User> {
```

Comment:
One of the requirements for this function is that tokens can only be used once. So when a token is 'valid', it should be deleted from the table.